### PR TITLE
Fix Infinispan Daily Builds

### PR DESCRIPTION
**Pull request by SudharakaP**
This is a fix for the Infinispan daily builds. The build is broken due to a bug in Infinispan it seems. When there's a single node infinispan cluster the discovery service seems to run into some problems. 

Fixes part of #10604

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by SudharakaP** [comment]
Close/Open for tests.

**Comment by SudharakaP** [comment]
Just to note the failure in CI is not related to this problem (its something to do with CI server caching in my opinion; correct me if I am wrong); I closed/reopened the issue and it passed; note that it still shows as failed due to the way GitHub actions operate; they don't replace the failed build but rather duplicate it :smile:

File: generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs [hunk 1]
```diff
@@ -448,13 +448,21 @@ public class CacheConfiguration<% if (cacheProvider === 'hazelcast') { %> implem
                     .clusterName("infinispan-<%=baseName%>-cluster").globalJmxStatistics()
                     .enabled(jHipsterProperties.getCache().getInfinispan().isStatsEnabled())
                     .allowDuplicateDomains(true).build();
+            } else if (discoveryClient.getInstances(registration.getServiceId()).size() == 0) {
+                return () -> GlobalConfigurationBuilder
+                .defaultClusteredBuilder().defaultCacheName("infinispan-store-cluster-cache").transport()
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs]: missing indent here

File: generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs [hunk 2]
```diff
@@ -448,13 +448,21 @@ public class CacheConfiguration<% if (cacheProvider === 'hazelcast') { %> implem
...
+                .transport(new JGroupsTransport())
+                .clusterName("infinispan-store-cluster").globalJmxStatistics()
+                .enabled(jHipsterProperties.getCache().getInfinispan().isStatsEnabled())
+                .allowDuplicateDomains(true).build();
+            } else {
+                return () -> GlobalConfigurationBuilder
+                .defaultClusteredBuilder().defaultCacheName("infinispan-store-cluster-cache").transport()
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs]: missing indent here too

File: generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs (hunk 1 above)
Comment (SudharakaP) [reply generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs]: Done. I wish there was a way these indentations are caught by the IDE. One of the many problems I was thinking about related to EJS. 🤔 🤔

File: generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs (hunk 2 above)
Comment (SudharakaP) [reply generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs]: Thanks. Done 😄

File: generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs [hunk 3]
```diff
@@ -448,13 +448,21 @@ public class CacheConfiguration<% if (cacheProvider === 'hazelcast') { %> implem
...
                     .enabled(jHipsterProperties.getCache().getInfinispan().isStatsEnabled())
                     .allowDuplicateDomains(true).build();
-            }
-            return () -> GlobalConfigurationBuilder
-                    .defaultClusteredBuilder().defaultCacheName("infinispan-<%=baseName%>-cluster-cache").transport()
+            } else if (discoveryClient.getInstances(registration.getServiceId()).size() == 0) {
+                return () -> GlobalConfigurationBuilder
+                    .defaultClusteredBuilder().defaultCacheName("infinispan-store-cluster-cache").transport()
+                    .transport(new JGroupsTransport())
+                    .clusterName("infinispan-store-cluster").globalJmxStatistics()
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs]: sorry for adding a new comment, just seen it after a 2nd review

Why did you change the clusterName ? maybe should we keep `infinispan-<%=baseName%>-cluster` ?

Comment (SudharakaP) [reply generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs]: @pascalgrimaud : Thank you for noticing and pointing this out. That was unintentional. I copied and pasted from my generated project; I should have noticed that. Sorry 😄
